inc/Api/Api.php

```diff
@@ -58,9 +58,19 @@ public function register_api_information() {
 				} elseif ( $sanitized['password'] === '' ) {
 					$response = array( 'error' => 'Please enter password!' );
 				} else {
+					// Get subdomain if entered as full url.
+					if ( filter_var( $sanitized['subdomain'], FILTER_VALIDATE_URL ) ) {
+						// Get host adress.
+						$url = wp_parse_url( $sanitized['subdomain']);
+						// Get only subdomain from host adress.
+						$doman = explode('.', $url['host'])[0];
```

Comment:
If user inserts URL that does not contain subdomain, for example `https://sendsmaily.net`, `sendsmaily` will become subdomain value. I think in such case user should be treated with error message.

Reply:
Fixed in https://github.com/sendsmaily/smaily-woocommerce-plugin/pull/27/commits/279bdd8f57579fcb459659b33854b32ac2f862ee

Reply:
There are some issues with the implementation:
1. the error message does not stop API call from being made;
2. checking if `$domain` equals to `sendsmaily` is bad, because there might be a subdomain with that value. I'd rather count the number of array items returned by `explode` (it should be greater or equal to 3);
3. there is a typo in the `$domain` variable name (on line 66 where `$doman` variable is assigned);
4. it does not work when user inserts URL without `https://`, for example `abc.sendsmaily.dev`;

Reply:
Changed in https://github.com/sendsmaily/smaily-woocommerce-plugin/pull/27/commits/f882c0ad536188ccf371513eb45dd2e92e71a901